src/Shared/ChangeWaves.cs

```diff
@@ -0,0 +1,45 @@
+using Microsoft.Build.Shared;
+using System;
+
+namespace Microsoft.Build.Utilities
+{
+    public class ChangeWaves
+    {
+        public const string Wave16_8 = "16.8";
+        public const string Wave16_10 = "16.10";
+        public const string Wave17_0 = "17.0";
+
+        /// <summary>
+        /// Compares version against the MSBuildChangeWave environment variable.
+        /// Version MUST be of the format: "xx.yy".
+        /// </summary>
+        /// <param name="wave">The version to compare.</param>
+        /// <returns>A bool indicating whether the version is enabled.</returns>
+        public static bool IsChangeWaveEnabled(string wave)
+        {
+            // This is opt out behavior, all waves are enabled by default.
+            // If version is invalid, 
+            if (string.IsNullOrEmpty(Traits.Instance.MSBuildChangeWave))
+            {
+                return true;
+            }
+
+            Version currentDisabledWave;
+
+            if (!Version.TryParse(Traits.Instance.MSBuildChangeWave, out currentDisabledWave))
```

Comment:
All this version parsing might get noticeable, especially if the under-the-wave features are on hotter paths. Consider storing the version constants directly as `Version` instead of strings.